### PR TITLE
Dask final

### PR DESCRIPTION
**Pull request by swainn**
The moment you've all been waiting for... the DaskJob type is finally ready. This pull request also begins the process of removing Python 2 support from Tethys Platform. I've created a new branch for Tethys 3 development until 2.1 is released. Then we can merge it into master.

Dask Job Type

NOTE: Python 3 ONLY.

tethys_compute
* Reorganized models into separate modules and packages.
* Added DaskJob model to handle Dask jobs.
* Added DaskSerializedField to automatically serialize and deserialize using Dask serialization.
* Refactored Scheduler model to be base class.
* Added two specific Scheduler subclasses: CondorScheduler and DaskScheduler.
* Added new TethysJob status: Results-Ready (RES).
* Added setter for TethysJob.status.
* Added Admin page for CondorScheduler and DaskScheudlers.
* Embedded Dask Bokeh dashboard in Admin pages DaskSchedulers.
* Added DASK job type to job_manager.
* Added new methods for creating Dask and Condor Schedulers (create_condor_scheduler and create_dask_scheduler) to scheduler_manager.
* Refactored scheduler_manager.create_scheduler to have a scheduler_type paramater to support creation of Condor and Dask Schedulers.

tethys_gizmos
* Updated jobs_table gizmos to show an embedded status graph from the Bokeh dashboard app as a details view.
* Added DAG graph view for CondorWorkflow jobs to jobs_table gizmo.
* Added new views for handling the update of the DAG graph rows and the Dask status graph rows.
* Moved jobs table demo in gizmo showcase to separate page.

tethys_apps
* Added new view for handling dask job status updates from dask scheduler (update_dask_job_status).
* Refactored scheduler create command in CLI into two sub-commands: dask-create and condor-create.

tethys_sdk
* Added new create_scheduler endpoints to compute API.
* Added TethysJob models to jobs API.

tethys_services
* Fixed Python 2/3 compatibility bugs.

Miscellaneous
* Added dask and tethys_dask_scheduler conda dependencies.
* Removed environment_py2.yml and refactored Dockerfile, install scripts, read the docs config, and travis config to only use Python 3.

Documentation
* New tutorial that illustrates how to use the DaskJob type.
* New Job Type documentation for DaskJob in SDK docs.

Tests
* Updated and added tests to maintain 99% coverage.
* Linted code using Flake8 to enforce PEP-8 code style.

### CODE REVIEW

**Comment by swainn** [comment]
I should also add that this includes a few additional things that were added by Aquaveo during our normal development. Mostly bug fixes and tweaks to the Docker build. Some of those commits could probably be merged into the master branch (everything up to the Dask Job Type).

**Comment by swainn** [comment]
@sdc50 is there any reason you are waiting to merge this?

File: tox.ini [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [flake8]
 max-line-length = 120
```
Comment (sdc50) [inline tox.ini]: Does Travis use this file?

File: tethys_sdk/jobs.py [hunk 2]
```diff
@@ -20,6 +20,9 @@
     # CondorWorkflowDataJobTemplate,
     # CondorWorkflowFinalTemplate,
 )
+from tethys_compute.models import (
+    BasicJob, CondorJob, CondorWorkflow, CondorWorkflowJobNode, DaskJob
+)
 
-# Depricated imports
```
Comment (sdc50) [inline tethys_sdk/jobs.py]: Guilty

File: .appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+version: 1.6.4-{build}
```
[the remark this answers is not in the document]
Comment (swainn) [reply .appveyor.yml]: Ah, yes. I think what happened there is I intended to configure Tethys to run tests for Windows, so I copied the tethys_dataset_services config as a starting point. I'll remove it.

File: scripts/install_tethys.bat [hunk 4]
```diff
@@ -350,7 +344,6 @@ ECHO     -b, --branch [BRANCH_NAME]          Branch to checkout from version con
 ECHO     -c, --conda-home [PATH]             Path to conda home directory where Miniconda will be installed. Default is %%TETHYS_HOME%%\miniconda.
 ECHO     -C, --conda-exe [PATH]              Path to Miniconda installer executable. Default is '.\Miniconda3-latest-Windows-x86_64.exe'.
 ECHO     -n, --conda-env-name [NAME]         Name for tethys conda environment. Default is 'tethys'.
-ECHO     --python-version [PYTHON_VERSION]   Main python version to install tethys environment into (2 or 3). Default is 2.
```
[the remark this answers is not in the document]
Comment (swainn) [reply scripts/install_tethys.bat]: Yeah... My thought on that was to not merge with master, but to cherry pick from master... and then force push over master. But I'd rather handle that after we release 2.1.

File: tox.ini (hunk 1 above)
Comment (swainn) [reply tox.ini]: stickler uses it for linting.

File: tests/unit_tests/test_tethys_apps/test_urls.py [hunk 5]
```diff
@@ -24,17 +24,17 @@ def test_urls(self):
...
-        self.assertEqual('/apps/test-app/foo/bar/', url)
-        self.assertEquals('home', resolver.func.__name__)
-        self.assertEqual('tethys_apps.tethysapp.test_app.controllers', resolver.func.__module__)
-
-        url = reverse('test_app:home', kwargs={'var1': 'foo', 'var2': 'bar'})
-        resolver = resolve(url)
-        self.assertEqual('/apps/test-app/foo/bar/', url)
-        self.assertEquals('home', resolver.func.__name__)
-        self.assertEqual('tethys_apps.tethysapp.test_app.controllers', resolver.func.__module__)
+        # url = reverse('test_app:home', args=['foo', 'bar'])
```
[the remark this answers is not in the document]
Comment (swainn) [reply tests/unit_tests/test_tethys_apps/test_urls.py]: I'm all for removing commented out code. Let your version software save it for you. Not sure why it is commented out.

File: tethys_sdk/jobs.py (hunk 2 above)
Comment (swainn) [reply tethys_sdk/jobs.py]: Haha

File: tests/unit_tests/test_tethys_apps/test_urls.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (swainn) [reply tests/unit_tests/test_tethys_apps/test_urls.py]: Is this the only case you found? I'm not sure how to systematically find and remove all cases of this.

File: .appveyor.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (swainn) [reply .appveyor.yml]: I removed it in the most recent commit.

File: tests/unit_tests/test_tethys_apps/test_urls.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (swainn) [reply tests/unit_tests/test_tethys_apps/test_urls.py]: After reviewing, it looks like someone got lazy and commented out part of the test rather then fixing it. So I fixed it in the latest commit.

**Comment by swainn** [comment]
Btw... I goofed and accidentally pushed the cesium final branch here, which made github think this was merged. I have reverted the master30, but it didn't seem to mark this as unmerged... Sorry.

**Pull request merged**